app/services/power_manager/base.rb

```diff
@@ -9,6 +9,21 @@ def self.method_missing(method, *args)
       raise ::Foreman::Exception.new(N_("Invalid power state request: %{action}, supported actions are %{supported}"), { :action => action, :supported => SUPPORTED_ACTIONS })
     end
 
+    SUPPORTED_ACTIONS.each do |method|
+      define_method method do # def start
+        action_entry = action_map[method.to_sym] || {}
```

Comment:
I have expanded `#action_map` format from `:key => 'method_to_call'` to this:
``` Ruby
{
  :key => {
    :action => 'method_to_call',
    :output => 'transform_output'
  }
}
```
This will allow us to override the output transformation too. For compatibility (and simplicity) reasons I have left the option to define the hash using old style too, but I normalize it to the new format for simpler code.